composer/trainer/trainer.py

```diff
@@ -912,14 +912,17 @@ def _spin_dataloaders(self):
         # so it does not affect any other RNG reads
         for evaluator in self.state.evaluators:
             dataloader = evaluator.dataloader.dataloader
-            if isinstance(dataloader.sampler, torch.utils.data.DistributedSampler):
+            # FFCV dataloaders use their own sampling strategy
+            if isinstance(dataloader, torch.utils.data.DataLoader) and isinstance(dataloader.sampler,
+                                                                                  torch.utils.data.DistributedSampler):
                 dataloader.sampler.set_epoch(0)
             for _ in dataloader:
                 break
 
         # spin the train dataloader's sampler to get to the state of the desired epoch
         for epoch in range(int(self.state.timer.epoch)):
-            if isinstance(self.state.train_dataloader.sampler, torch.utils.data.DistributedSampler):
+            if hasattr(self.state.train_dataloader, "sampler") and isinstance(self.state.train_dataloader.sampler,
```

Comment:
Can you add a `# TODO` in the comments, saying that the hasattr check will be removed in #424?

Reply:
will do.